command.go

```diff
@@ -553,6 +590,78 @@ func (inv *Invocation) with(fn func(*Invocation)) *Invocation {
 	return &i2
 }
 
+func (inv *Invocation) doCompletions() []string {
+	prev, cur := inv.curWords()
+	inv.CurWord = cur
+	// If the current word is a flag set using `=`, use it's handler
+	if strings.HasPrefix(cur, "--") && strings.Contains(cur, "=") {
+		if out := inv.equalsFlagCompletions(cur); out != nil {
+			return out
+		}
+	}
+	// If the previous word is a flag, then we're writing it's value
+	// and we should check it's handler
+	if strings.HasPrefix(prev, "--") {
+		if out := inv.flagCompletions(prev); out != nil {
+			return out
+		}
+	}
+	// If the current word is the command, auto-complete it so the shell moves the cursor
+	if inv.Command.Name() == inv.CurWord {
+		return []string{inv.Command.Name()}
+	}
+	var completions []string
+
+	if inv.Command.CompletionHandler != nil {
+		completions = append(completions, inv.Command.CompletionHandler(inv)...)
+	}
+
+	completions = append(completions, DefaultCompletionHandler(inv)...)
+
+	return completions
+}
+
+func (inv *Invocation) flagCompletions(word string) []string {
+	return inv.doFlagCompletions("", word)
+}
+
+func (inv *Invocation) equalsFlagCompletions(word string) []string {
```

Comment:
Really hard to tell what this function does by looking at its signature. Typically a function named `equals...` returns a boolean and is checking equality.

Would make more sense as `completeEqualFlags`